PurchasesCoreSwift/Purchasing/StoreKitRequestFetcher.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  StoreKitRequestFetcher.swift
+//  PurchasesCoreSwift
+//
+//  Created by Andrés Boedo on 6/29/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+import StoreKit
+
+// todo: make internal
+@objc(RCReceiptRefreshRequestFactory) public class ReceiptRefreshRequestFactory: NSObject {
+
+    @objc public func receiptRefreshRequest() -> SKReceiptRefreshRequest {
```

Comment:
should this return type be the more generic SKRequest? otherwise is a factory necessary? 

just trying to understand what's actually going on here, not a migration question...

Reply:
yeah, the trick here is that although we do reference `SKRequest` directly, when testing, we use a [mock factory](https://github.com/RevenueCat/purchases-ios/blob/main/PurchasesTests/Purchasing/StoreKitRequestFetcherTests.swift#L34), which returns mock [receipt refresh requests](https://github.com/RevenueCat/purchases-ios/blob/main/PurchasesTests/Purchasing/StoreKitRequestFetcherTests.swift#L19). 

It's a inherit-and-replace approach. An alternative approach would be to have a generic request type, and have the `SKReceiptRefreshRequest` conform to that. I _think_ obj-c would allow for us to extend `SKReceiptRefreshRequest` to add conformance to the protocol, but this approach might be simpler

Reply:
as for `SKRequest` vs `SKReceiptRefreshRequest`, `SKRequest` would also cover `SKProductsRequest`, but the `RCStoreKitRequestFetcher` should never be compatible with those since it can't handle them. 
In other words, making this the generic `SKRequest` would violate the Liskov substitution principle, since a factory would validly be able to return an `SKProductsRequest`, but the class would no longer work as intended if it did.

Reply:
all of that said, we now have two factories, one for receipt refresh requests and one for products requests, which we could easily unify into one. I didn't do it here because I wanted the migration to introduce as little new code as possible